frontend/src/app/signup/page.tsx

```diff
@@ -0,0 +1,93 @@
+"use client";
+
+import Link from "next/link";
+import React from "react";
+import { useFormik } from "formik";
+import { SignUpFormValues } from "@utils/interfaces";
+import { Heading } from "@ui/Heading";
+import { Button } from "@ui/Button";
+
+const SignUpForm = () => {
+  const { values, handleChange, handleBlur, errors, touched, handleSubmit } =
+    useFormik<SignUpFormValues>({
+      initialValues: {
+        username: "",
+        password: "",
+        confirmPassword: "",
+      },
+
+      onSubmit: (values) => {
+        alert(JSON.stringify(values, null, 2));
+      },
+    });
+
+  return (
+    <section className="grid content-center max-w-full">
+      <h1 className={Heading({ align: "center", size: "small" })}>Sign up</h1>
+      <p className="mt-10 text-center">Enter your username and password</p>
+      <form onSubmit={handleSubmit} className="mt-3 w-96">
+        <div>
+          <input
+            type="text"
+            name="username"
+            id="username"
+            className="form_input"
+            placeholder="Username"
+            value={values.username}
+            onChange={handleChange}
+            onBlur={handleBlur}
+          />
+          {touched.username && errors.username ? (
+            <div>{errors.username}</div>
+          ) : null}
+        </div>
+        <div>
+          <input
+            type="password"
+            name="password"
+            id="password"
+            className="form_input"
+            placeholder="Password"
+            value={values.password}
+            onChange={handleChange}
+            onBlur={handleBlur}
+          />
+          {touched.password && errors.password ? (
+            <div>{errors.password}</div>
+          ) : null}
+        </div>
+        <div>
+          <input
+            type="password"
+            name="confirmPassword"
+            id="password"
+            className="form_input"
+            placeholder="Confirm Password"
+            value={values.confirmPassword}
+            onChange={handleChange}
+            onBlur={handleBlur}
+          />
+          {touched.confirmPassword && errors.confirmPassword ? (
+            <div>{errors.confirmPassword}</div>
+          ) : null}
+        </div>
+        <Button
+          type="submit"
+          intent="primary"
+          size="medium"
+          className="text-white w-full mt-5"
+        >
+          Sign up
+        </Button>
```

Comment:
Just a tiny thing to consider, In the Button component we already have a 'text-white w-full' class applied if you set the size 'large'. so, you don't need to add an extra class name to the button component.

Reply:
Thank you for the feedback! I'll remove the class name 'text-white w-full' from Button component since its already applied in large size

Reply:
Removed the extra classes